File: src/main/java/kitchenpos/product/tobe/domain/repository/ProductRepository.java
package kitchenpos.product.tobe.domain.repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import kitchenpos.product.tobe.domain.entity.Product;

public interface ProductRepository {

    Product save(Product product);

    Optional<Product> findById(UUID id);

    List<Product> findAll();

    List<Product> findAllByIdIn(List<UUID> ids);
}

File: src/main/kotlin/kitchenpos/tobe/product/domain/repository/ProductRepository.kt
package kitchenpos.tobe.product.domain.repository

import kitchenpos.tobe.product.domain.entity.ProductV2
import java.util.*

interface ProductRepository {
    fun save(productV2: ProductV2): ProductV2

    fun findById(id: UUID): Optional<ProductV2>

    fun findAll(): List<ProductV2>

    fun findAllByIdIn(ids: List<UUID>): List<ProductV2>
}
